Panic: Segmentation violation running --type-aware

Hi!

I experimented with running oxlint on the [bitwarden/clients](https://github.com/bitwarden/clients/tree/anders-oxlint) repo.
oxlint itself runs nicely, although without rxjs/angular rules.

However, when running with `--type-aware` we get the following seg fault.
I've tracked it down to be beacuse of `typescript/no-misused-promises`. Without that rule the call succeeds.

Not sure there's anything tsgolint can implement or if needs to be a upstream `typescript-go` fix.

```
❯ npm run oxlint:types

> @bitwarden/clients@0.0.0 oxlint:types
> oxlint --type-aware

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x100 pc=0x1025f0e00]

goroutine 328497 [running]:
github.com/microsoft/typescript-go/internal/checker.(*Checker).getBaseTypes(0x140366d2008, 0x1403f39bc00)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:18574 +0x30
github.com/microsoft/typescript-go/internal/checker.(*Checker).isPropertyDeclaredInAncestorClass(0x140366d2008, 0x140312dd1f8)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:11378 +0x70
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkPropertyNotUsedBeforeDeclaration(0x140366d2008, 0x140312dd1f8, 0x1403978c5f0?, 0x1403978c5f0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:11358 +0x2e4
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkPropertyAccessExpressionOrQualifiedName(0x140366d2008, 0x1403957c840, 0x14039570738, 0x1403f399b90, 0x1403978c5f0, 0x0, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:11022 +0x794
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkPropertyAccessExpression(0x140366d2008, 0x1403957c840, 0x0, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:10912 +0x7c
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionWorker(0x140366d2008, 0x1403957c840, 0x0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7537 +0x320
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionEx(0x140366d2008, 0x1403957c840, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7334 +0x50
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpression(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7327
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkNonNullExpression(0x140366d2008, 0x1403957c840)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7179 +0x28
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkPropertyAccessExpression(0x140366d2008, 0x1403957c898, 0x0, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:10912 +0x48
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionWorker(0x140366d2008, 0x1403957c898, 0x0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7537 +0x320
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionEx(0x140366d2008, 0x1403957c898, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7334 +0x50
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpression(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7327
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkNonNullExpression(0x140366d2008, 0x1403957c898)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7179 +0x28
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkPropertyAccessExpression(0x140366d2008, 0x14039343600, 0x40, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:10912 +0x48
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionWorker(0x140366d2008, 0x14039343600, 0x386db878?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7537 +0x320
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionEx(0x140366d2008, 0x14039343600, 0x40)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7334 +0x50
github.com/microsoft/typescript-go/internal/checker.(*Checker).getTypeOfExpression(0x140366d2008, 0x14039343600)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:7121 +0x68
github.com/microsoft/typescript-go/internal/checker.(*Checker).getRegularTypeOfExpression(0x140366d2008, 0x140237c8db0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:31039 +0xc0
github.com/microsoft/typescript-go/internal/checker.(*Checker).getTypeOfNode(0x140366d2008, 0x14039343600)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:30876 +0x4d0
github.com/microsoft/typescript-go/internal/checker.(*Checker).GetTypeAtLocation(0x0?, 0x0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/checker/checker.go:31080 +0x1c
github.com/typescript-eslint/tsgolint/internal/rules/no_misused_promises.init.func1.15(0x1403957c7e8)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_misused_promises/no_misused_promises.go:398 +0xa0
github.com/typescript-eslint/tsgolint/internal/rules/no_misused_promises.init.func1.16(0x1403957c7e8)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_misused_promises/no_misused_promises.go:473 +0x34
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.6(0xd6, 0x1403957c7e8)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:295 +0x6c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.8(0x1403957c7e8)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:342 +0x48
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*PropertyDeclaration).ForEachChild(0x14039570a80, 0x1400aedb820)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:5981 +0x110
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x140396b00ad?, 0x14039570a80?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.8(0x14039570a80)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:357 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*ClassLikeBase).ForEachChild(0x14038fd7c58, 0x1400aedb820)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:4162 +0x134
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x400000108?, 0x14038fd7c20?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2.8(0x14038fd7c20)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:357 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*SourceFile).ForEachChild(0x140396b7808, 0x1400aedb820)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:10862 +0xa4
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x102d40500?, 0x14036846f30?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func2()
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:365 +0xd0
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/core/workgroup.go:40 +0x24
sync.(*WaitGroup).Go.func1()
        /opt/hostedtoolcache/go/1.25.0/x64/src/sync/waitgroup.go:239 +0x4c
created by sync.(*WaitGroup).Go in goroutine 1
        /opt/hostedtoolcache/go/1.25.0/x64/src/sync/waitgroup.go:237 +0x70
```

--- a/patches/0007-fix-nil-interface-type-crash-in-getBaseTypes.patch
+++ b/patches/0007-fix-nil-interface-type-crash-in-getBaseTypes.patch
@@ -1,0 +1,36 @@
+From 7cf3b2d467e97607365d4845c6df7a20006dcf23 Mon Sep 17 00:00:00 2001
+From: Cameron Clark <cameron.clark@hey.com>
+Date: Thu, 11 Dec 2025 21:35:43 +0000
+Subject: [PATCH] fix: nil interface type crash in getBaseTypes
+
+---
+ internal/checker/checker.go | 6 ++++++
+ 1 file changed, 6 insertions(+)
+
+diff --git a/internal/checker/checker.go b/internal/checker/checker.go
+index 047517630..2c2b3b046 100644
+--- a/internal/checker/checker.go
++++ b/internal/checker/checker.go
+@@ -11381,6 +11381,9 @@ func (c *Checker) isPropertyDeclaredInAncestorClass(prop *ast.Symbol) bool {
+ 			return false
+ 		}
+ 		classType = baseTypes[0]
++		if classType.symbol == nil || classType.symbol.Flags&(ast.SymbolFlagsClass|ast.SymbolFlagsInterface) == 0 {
++			return false
++		}
+ 		superProperty := c.getPropertyOfType(classType, prop.Name)
+ 		if superProperty != nil && superProperty.ValueDeclaration != nil {
+ 			return true
+@@ -18562,6 +18565,9 @@ func findIndexInfo(indexInfos []*IndexInfo, keyType *Type) *IndexInfo {
+ 
+ func (c *Checker) getBaseTypes(t *Type) []*Type {
+ 	data := t.AsInterfaceType()
++	if data == nil {
++		return nil
++	}
+ 	if !data.baseTypesResolved {
+ 		if !c.pushTypeResolution(t, TypeSystemPropertyNameResolvedBaseTypes) {
+ 			return data.resolvedBaseTypes
+-- 
+2.50.1 (Apple Git-155)
+

